google/api_core/path_template.py

```diff
@@ -298,6 +298,15 @@ def transcode(http_options, message=None, **request_kwargs):
         # Assign body and query params
         body = http_option.get("body")
 
+        # gapic-generator-python appends an underscore to field names
+        # that collide with python keywords.
+        # `_` is stripped away as it is not possible to
+        # natively define a field with a trailing underscore in protobuf.
+        # See related issue
+        # https://github.com/googleapis/python-api-core/issues/227
+        if isinstance(leftovers, dict):
```

Comment:
`leftovers` can also be a `proto.Message`, as the original `transcoded_value` ([code](https://github.com/googleapis/python-api-core/blob/main/google/api_core/path_template.py#L274)). 

Shouldn't we take care of it too?

Reply:
Oh I see, you're handling it in lines 320-328. I think you can move this logic up here. `leftovers` is either a message or `dict`, so you can do this preprocessing together. It will be more readable too.

Reply:
Fixed in [5504d0a](https://github.com/googleapis/python-api-core/pull/489/commits/5504d0ad2112f328d42af3eba5ed3a0a98c2a292)